src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,377 @@
+            trip.queryParams = query;
+            Map<String, String> params = ItineraryUtils.excludeRealtime(trip.parseQueryParams());
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    /**
+     * Check that a given trip is modified with an OTP itinerary that exists,
+     * right now based on the index of the selected itinerary in the UI.
+     */
+    @Test
+    public void testUpdateTripWithVerifiedItinerary() throws URISyntaxException {
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "testUpdateTripWithVerifiedItinerary";
+        trip.queryParams = BASE_QUERY + "&ui_activeItinerary=1";
+        trip.itinerary = null;
+
+        List<Itinerary> itineraries = otpDispatcherPlanResponse.getResponse().plan.itineraries;
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createSameDayTestCases")
+    void testIsSameDay(SameDayTestCase testCase) throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip(testCase.isArrival);
+        // The time zone for trip (and testCase.tripTime) is US Eastern per trip location.
+        ZoneId zoneId = trip.timezoneForTargetLocation();
+
+        Itinerary itinerary = simpleItinerary(testCase.tripTime, testCase.isArrival);
+        Assertions.assertEquals(
+            testCase.shouldBeSameDay,
+            ItineraryUtils.isSameDay(itinerary, QUERY_DATE, testCase.timeOfDay, zoneId, testCase.isArrival),
+            testCase.getMessage(zoneId)
+        );
+    }
+
+    private static List<SameDayTestCase> createSameDayTestCases() {
+        return List.of(
+            // Same-day departures
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__03_00_00_EDT, false, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__23_59_59_EDT, false, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__02_59_59_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_12__03_00_00_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_12__23_59_59_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_13__02_59_59_EDT, false, true),
+
+            // Not same-day departures
+            new SameDayTestCase(QUERY_TIME, _2020_08_12__23_59_59_EDT, false, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__02_59_59_EDT, false, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__03_00_00_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_13__03_00_00_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_13__23_59_59_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_14__02_59_59_EDT, false, false),
+
+            // Same-day arrivals
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__03_00_00_EDT, true, true),
```

Comment:
This one is kind of tricky. Since it does end on the desired date, I suppose it makes sense. However, it implies that the trip has started on the previous day. If this ever comes up it's probably due to some really infrequent routes that run at odd times, so I guess it's fine to keep for now.